lib/cn/csched_sp3_work.c

```diff
@@ -379,15 +379,13 @@ sp3_work_leaf_len(
                     runlen = runlen_min;
                 }
             }
-
-            return runlen;
         }
 
         /* Don't let lightweight nodes grow too long.  For the most part this only
          * applies to "index" nodes (i.e., nodes where the values are much smaller
          * than the keys).
          */
-        if (cn_ns_kvsets(&stats) > runlen_min &&
+        else if (cn_ns_kvsets(&stats) > runlen_min &&
```

Comment:
The `else if` not being on the same line as the previous closing brace is not ideal